stacks-signer/src/runloop.rs

```diff
@@ -175,8 +194,40 @@ impl<C: Coordinatable> RunLoop<C> {
         let inbound_messages: Vec<Packet> = event
             .modified_slots
             .iter()
+            // intercept and consume ping packets
+            .filter(|chunk| {
```

Comment:
Thanks for cleaning this up a bit.  Can you also put this `filter()` and `filter_map()` logic into its own method?